app/lib/frontend/handlers.dart

```diff
@@ -356,20 +361,28 @@ Future<shelf.Response> packageVersionHandlerHtml(
       }
     }
 
+    final Future<AnalysisReport> analysisReportFuture = useAnalyzerService
+        ? analyzerClient.getAnalysisData(packageName, selectedVersion.version)
+        : new Future.value();
+
     final versionDownloadUrls =
         await Future.wait(first10Versions.map((PackageVersion version) {
       return backend.downloadUrl(packageName, version.version);
     }).toList());
 
+    final AnalysisReport report = await analysisReportFuture;
```

Comment:
To run it in concurrently to the async construction of `versionDownloadUrls` on the next few lines. It reduces a bit of latency, and doesn't seem to make things complicated.